### PR TITLE
Updating to allow borrow direct and ILL for Resource shared items

### LINKED ISSUE #971: Items in Resource Sharing Request Process Type Should provide Resource Sharing Request Options
Example: https://catalog-alma-qa.princeton.edu/requests/9977664533506421?mfhd=22109013720006421&source=pulsearch

Items in this status have been requested by a resource sharing partner, PU patrons interested in one these items should have their request directed to Borrow Direct/ILLiad. 

To find current examples in Alma's backend you can search by Physical Items -> Process Type = "Resource Sharing Request"

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #971

### CODE REVIEW

**Review by kevinreiss — COMMENTED** [review]
This looks good. I think the failed specs are just an issue with the patron fixtures.

File: app/models/requests/requestable.rb [hunk 1]
```diff
@@ -281,6 +281,10 @@ def cul_music?
       item[:collection_code].present? && item[:collection_code] == 'MR'
     end
 
+    def resource_shared?
```
Comment (kevinreiss) [inline app/models/requests/requestable.rb:284]: @mzelesky will any items that find themselves in various resource sharing workflows always fine themselves in the RES_SHARE library?

Comment (mzelesky) [reply app/models/requests/requestable.rb:284]: Do you mean our own items that are loaned to other institutions?

**Comment by carolyncole** [comment]
@kevinreiss this is finally green...  I did not notice I was creating the patron in a slightly different way...

**Review by kevinreiss — APPROVED** [review]

**Pull request merged**